src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -878,7 +878,7 @@ private static void ensureVisible(ClassLoader ld, Class<?> c) {
             }
             if (type != c) {
                 throw new IllegalArgumentException(c.getName() +
-                        " referenced from a method is not visible from class loader");
+                        " referenced from a method is not visible from class loader: " + ld);
```

Comment:
The ClassLoader string representation may or may not be useful here. One suggestion is to use the class loader name (ld.getName()) if not null, otherwise Objects.toIdentityString(ld).

Reply:
I also think it's helpful to print the name of class loader if present, or the identity string as Alan suggests.

Reply:
@AlanBateman @mlchung - Thanks for reviewing. I've incorporated review comments wherein using <loader-name>@<id> if not null, otherwise Objects.toIdentityString(ld). PTAL.